src/Storages/RabbitMQ/StorageRabbitMQ.cpp

```diff
@@ -882,19 +896,8 @@ bool StorageRabbitMQ::streamToViews()
             {
                 /// Iterate loop to activate error callbacks if they happened
                 event_handler->iterateLoop();
-
-                if (event_handler->connectionRunning())
-                {
-                    /* Almost any error with channel will lead to connection closure, but if so happens that channel errored and
-                     * connection is not closed - also need to restore channels
-                     */
-                    if (!stream->as<RabbitMQBlockInputStream>()->needChannelUpdate())
-                        stream->as<RabbitMQBlockInputStream>()->updateChannel();
-                }
-                else
-                {
+                if (!event_handler->connectionRunning())
```

Comment:
Don't understand these changes.

Reply:
It is just those checks (and channel update if needed) were moved to another place. They now will be done in readPrefix when buffer is popped. It is more optimal.